nautobot/core/graphql/types.py

```diff
@@ -24,6 +26,23 @@ class Meta:
         model = ContentType
 
 
+class Date(graphene.Date):
+    """
+    Overriding the default serialize method from https://github.com/graphql-python/graphene/blob/master/graphene/types/datetime.py
+    to handle the case where the date object is passed as a str object.
+    """
+
+    @staticmethod
+    def serialize(date):
+        if isinstance(date, datetime.datetime):
+            date = date.date()
+        if not isinstance(date, datetime.date) and isinstance(date, str):
+            date = datetime.datetime.strptime(date, "%Y-%m-%d").date()
+        if not isinstance(date, datetime.date):
+            raise AssertionError('Received not compatible date "{}"'.format(repr(date)))
+        return date.isoformat()
```

Comment:
Just to make sure I follow - if we're given a string, we parse it into a datetime.date, then re-export it as a new string? Any way we can avoid that round-trip?